### PR TITLE
Remove mocha in favor of jest

### PR DESCRIPTION
**Pull request by arturovt**
Chai+Mocha is a bit outdated solution and doesn't have reach testing functionality compared to Jest. I was struggling e.g. to setup chai-spies to test spies with chai.

### CODE REVIEW

File: tsconfig.spec.json [hunk 1]
```diff
@@ -3,7 +3,7 @@
   "compilerOptions": {
     "strict": false,
     "module": "CommonJS",
-    "types": ["node", "mocha", "chai", "chai-spies"]
+    "types": ["node", "jest"]
```
Comment (nateplusplus) [inline tsconfig.spec.json:6]: Nice! Replacing 3 dependencies with one!

**Comment by nateplusplus** [comment]
This sounds good. Yeah I know jest is far more popular now. I've just had more experience with mocha so I started this project with it, but now that it is growing I think Jest is a good way forward. Looking at your work, it seems like Jest is very similar to Mocha/Chai so I can dig it!

Just one issue: It looks like there is a problem with the GitHub CI/CD Unit Test checks. Not sure if you are able to see this result? https://github.com/nateplusplus/pushin/runs/6049158482?check_suite_focus=true

**Comment by nateplusplus** [comment]
@arturovt looking good now! I noticed this is still a draft. Should I merge it in or wait?

**Comment by arturovt** [comment]
Marked as ready.

**Pull request merged**